modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2DeviceFlowTestCase.java

```diff
@@ -233,45 +221,35 @@ public void testSendApprovalPost() throws Exception {
     @Test(groups = "wso2.is", description = "Send approval post request", dependsOnMethods = "testSendApprovalPost")
     public void testTokenRequest() throws Exception {
 
-        List<NameValuePair> urlParameters = new ArrayList<>();
-        urlParameters.add(new BasicNameValuePair(OAuth2Constant.GRANT_TYPE_NAME,
-                "urn:ietf:params:oauth:grant-type:device_code"));
-        urlParameters.add(new BasicNameValuePair(DEVICE_CODE, deviceCode));
-        urlParameters.add(new BasicNameValuePair(CLIENT_ID_PARAM, consumerKey));
-        HttpPost request = new HttpPost(OAuth2Constant.ACCESS_TOKEN_ENDPOINT);
-        request.setHeader(CommonConstants.USER_AGENT_HEADER, OAuth2Constant.USER_AGENT);
-        request.setEntity(new UrlEncodedFormEntity(urlParameters));
-        HttpResponse response = client.execute(request);
-        BufferedReader rd = new BufferedReader(new InputStreamReader(response.getEntity().getContent()));
-        Object obj = JSONValue.parse(rd);
-        String accessToken = ((JSONObject) obj).get("access_token").toString();
+        Thread.sleep(5000); // Sleep 5 seconds before sending the token request
```

Comment:
better make the comment bit more descriptive to explain the requirement to sleep.